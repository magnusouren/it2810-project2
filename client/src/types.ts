export interface User {
  name: string;
  id: string;
  loginState: boolean;
  ratings?: Rating[];
  watchlist?: Movie[];
}

/**
 * The Movie type is based on the API response from
 * https://developers.themoviedb.org/3/movies/get-popular-movies
 * The API response is a bit more complex, but we only need these fields.
 */
export interface Movie {
  adult: boolean;
  backdrop_path: string;
  genre_ids: number[];
  _id: number;
  original_language: string;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string;
  release_date: string;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
}

/**
 * The Category type is based on the API response from
 * https://developers.themoviedb.org/3/genres/get-movie-list
 */
export interface Category {
  id: number;
  name: string;
}

export interface Rating {
  movieId: number;
  rating: number;
}

export type AlphabeticalSort = 'a-z' | 'z-a' | '';

export type RatingSort = 'h-l' | 'l-h' | '';
