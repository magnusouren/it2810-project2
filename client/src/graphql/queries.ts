import { gql } from '@apollo/client';

export const GET_MOVIES = gql`
  query getMovies($page: Int!, $userId: String) {
    getMovies(page: $page, userID: $userId) {
      _id
      genre_ids {
        _id
        name
      }
      poster_path
      title
      isInWatchlist(userID: $userId)
    }
    getMovieCountByGenre(genreId: null)
  }
`;

export const GET_MOVIE = gql`
  query getMovieById($id: Int!, $userId: String) {
    getMovieById(id: $id) {
      _id
      backdrop_path
      genre_ids {
        _id
        name
      }
      overview
      popularity
      release_date
      title
      vote_average
      vote_count
      isInWatchlist(userID: $userId)
    }
  }
`;

export const GET_MOVIES_BY_GENRE = gql`
  query getMoviesByGenre($page: Int!, $genreId: Int!, $userId: String!) {
    getMoviesByGenre(page: $page, genreId: $genreId) {
      _id
      genre_ids {
        _id
        name
      }
      poster_path
      title
      isInWatchlist(userID: $userId)
    }
    getMovieCountByGenre(genreId: $genreId)
  }
`;

export const GET_MOVIES_BY_TITLE_AZ = gql`
  query getMoviesByTitleAZ($page: Int!, $order: String!, $genreId: Int, $userId: String) {
    getMoviesByTitleAZ(page: $page, order: $order, genreId: $genreId) {
      _id
      genre_ids {
        _id
        name
      }
      isInWatchlist(userID: $userId)
      poster_path
      title
    }
    getMovieCountByGenre(genreId: $genreId)
  }
`;

export const GET_MOVIES_BY_RATING = gql`
  query getMoviesByRating($page: Int!, $order: String!, $genreId: Int, $userId: String) {
    getMoviesByRating(page: $page, order: $order, genreId: $genreId) {
      _id
      genre_ids {
        _id
        name
      }
      poster_path
      title
      isInWatchlist(userID: $userId)
    }
    getMovieCountByGenre(genreId: $genreId)
  }
`;

export const GET_MOVIES_BY_TITLE = gql`
  query getMoviesByTitle($title: String!, $limit: Int!) {
    getMoviesByTitle(title: $title, limit: $limit) {
      _id
      title
    }
  }
`;

export const REMOVE_MOVIE_FROM_WATCHLIST = gql`
  mutation Mutation($userId: String!, $movieId: Int!) {
    removeMovieFromWatchlist(userID: $userId, movieID: $movieId) {
      userID
      movies {
        _id
      }
    }
  }
`;

export const ADD_MOVIE_TO_WATCHLIST = gql`
  mutation Mutation($userId: String!, $movieId: Int!) {
    addMovieToWatchlist(userID: $userId, movieID: $movieId) {
      userID
      movies {
        _id
      }
    }
  }
`;

export const GET_WATCHLIST_BY_USER_ID = gql`
  query Query($userId: String!, $page: Int!) {
    getWatchlistByUserID(userID: $userId, page: $page) {
      userID
      movies {
        _id
        title
        genre_ids {
          _id
          name
        }
        poster_path
        isInWatchlist(userID: $userId)
      }
    }
    getWatchlistCountByUserID(userID: $userId)
  }
`;

export const GET_MOVIE_RATING_WITH_USERID = gql`
  query getMovieRatingWithUserID($userID: String!, $movieID: Int!) {
    getMovieRatingWithUserID(userID: $userID, movieID: $movieID) {
      rating
    }
  }
`;

export const ADD_RATING = gql`
  mutation addRating($userID: String!, $movieID: Int!, $rating: Float!) {
    addRating(userID: $userID, movieID: $movieID, rating: $rating) {
      rating
    }
  }
`;

export const GET_GENRES = gql`
  query getGenres {
    getGenres {
      _id
      name
    }
  }
`;

export const GET_FILTER = gql`
  query FilterData {
    filter @client {
      sort
      genre
      page
    }
  }
`;

export const determineQueryAndVariables = (page: number, genre: string, sort: string, userId: string) => {
  const baseVariables = { page: page, userId: userId };
  const alphabeticalSort = sort === 'a-z' || sort === 'z-a';
  const ratingSort = sort === 'h-l' || sort === 'l-h';

  if (genre) {
    if (alphabeticalSort) {
      return {
        query: GET_MOVIES_BY_TITLE_AZ,
        variables: { ...baseVariables, genreId: genre, order: sort },
      };
    }
    if (ratingSort) {
      return {
        query: GET_MOVIES_BY_RATING,
        variables: { ...baseVariables, genreId: genre, order: sort },
      };
    }
    return {
      query: GET_MOVIES_BY_GENRE,
      variables: { ...baseVariables, genreId: genre },
    };
  }

  if (alphabeticalSort) {
    return {
      query: GET_MOVIES_BY_TITLE_AZ,
      variables: { ...baseVariables, order: sort },
    };
  }
  if (ratingSort) {
    return {
      query: GET_MOVIES_BY_RATING,
      variables: { ...baseVariables, order: sort },
    };
  }
  return {
    query: GET_MOVIES,
    variables: { ...baseVariables },
  };
};
