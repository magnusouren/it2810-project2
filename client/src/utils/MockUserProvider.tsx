import { FC, ReactNode, useState } from 'react';
import { vi } from 'vitest';

// Importer UserContext
import { UserContext } from '../context/UserContext'; // Importer UserContext
import { User } from '../types';

interface MockUserProviderProps {
  children: ReactNode;
  mockUser?: User;
  loginMock?: () => void;
  logoutMock?: () => void;
  deleteUserMock?: () => void;
  toggleDarkModeMock?: () => void;
}

const MockUserProvider: FC<MockUserProviderProps> = ({
  children,
  mockUser,
  loginMock = vi.fn(),
  logoutMock = vi.fn(),
  deleteUserMock = vi.fn(),
  toggleDarkModeMock = vi.fn(),
}) => {
  const [user, setUser] = useState<User | undefined>(mockUser);

  const login = loginMock;
  const logout = logoutMock;
  const deleteUser = deleteUserMock;
  const toggleDarkMode = toggleDarkModeMock;

  const value = {
    user,
    setUser,
    login,
    logout,
    deleteUser,
    darkMode: false,
    toggleDarkMode,
  };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

export default MockUserProvider;
