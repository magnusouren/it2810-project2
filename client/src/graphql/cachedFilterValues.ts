import { client } from '../main';
import { Sort } from '../types';
import { GET_FILTER } from './queries';

type FilterResponse = {
  filter: Filter | null;
};

type Filter = {
  sort: Sort;
  genre: string;
  page: number;
};

/**
 *
 * Gets the cached filter values.
 *
 * @returns {Filter}
 */
export const getCachedFilterValues = () => {
  const cache = client.cache;
  const filterResponse = cache.readQuery<FilterResponse>({
    query: GET_FILTER,
  });
  const filter: Filter = filterResponse?.filter || { sort: '', genre: '', page: 1 };

  return filter;
};

/**
 *
 * Sets the cached filter values.
 *
 * @param {Filter} filter
 */
export const setCachedFilterValues = ({ sort, genre, page }: Filter) => {
  const cache = client.cache;

  cache.writeQuery({
    query: GET_FILTER,
    data: {
      filter: {
        sort: sort,
        genre: genre,
        page: page,
      },
    },
  });
};
