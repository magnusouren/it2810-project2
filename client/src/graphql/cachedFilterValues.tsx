import { client } from '../main';
import { AlphabeticalSort, RatingSort } from '../types';
import { GET_FILTER } from './queries';

type FilterResponse = {
  filter: Filter | null;
};

type Filter = {
  alphabeticalSort: AlphabeticalSort;
  genre: string;
  ratingSort: RatingSort;
  page: number;
};

/**
 *
 * Gets the cached filter values.
 *
 * @returns {Filter}
 */
export const getCachedFilterValues = () => {
  const cache = client.cache;
  const filterResponse = cache.readQuery<FilterResponse>({
    query: GET_FILTER,
  });
  const filter: Filter = filterResponse?.filter || { alphabeticalSort: '', genre: '', ratingSort: '', page: 1 };

  return filter;
};

/**
 *
 * Sets the cached filter values.
 *
 * @param {Filter} filter
 */
export const setCachedFilterValues = ({ alphabeticalSort, genre, ratingSort, page }: Filter) => {
  const cache = client.cache;

  cache.writeQuery({
    query: GET_FILTER,
    data: {
      filter: {
        alphabeticalSort: alphabeticalSort,
        genre: genre,
        ratingSort: ratingSort,
        page: page,
      },
    },
  });
};
