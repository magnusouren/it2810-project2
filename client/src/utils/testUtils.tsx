import { ApolloClient, ApolloProvider, HttpLink, InMemoryCache } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { User } from '../types';
import MockUserProvider from './MockUserProvider';

const link = new HttpLink({
  uri: 'http://localhost:4000', // Endepunktet til din GraphQL server
});

const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: link,
});

type ProviderMocks = {
  loginMock?: () => void;
  logoutMock?: () => void;
  deleteUserMock?: () => void;
  toggleDarkModeMock?: () => void;
};

interface ProviderProps {
  child: React.ReactNode;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  mocks?: any[];
  mockUser?: User;
  providerMocks?: ProviderMocks;
}

// Function to render a component wrapped inside a browserrouter for testing purposes
export const renderWithProviders = ({ child, mockUser, mocks, providerMocks }: ProviderProps) =>
  render(
    <MockUserProvider mockUser={mockUser} {...providerMocks}>
      <ApolloProvider client={client}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <BrowserRouter>{child}</BrowserRouter>
        </MockedProvider>
      </ApolloProvider>
    </MockUserProvider>,
  );
