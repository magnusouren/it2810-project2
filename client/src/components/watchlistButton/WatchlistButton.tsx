import { useMutation } from '@apollo/client';
import { BookmarkAdd, BookmarkRemove } from '@mui/icons-material';
import { ToggleButton } from '@mui/material';
import { FC, memo, useState } from 'react';

import { ADD_MOVIE_TO_WATCHLIST, REMOVE_MOVIE_FROM_WATCHLIST } from '../../graphql/queries';
import { Movie, User } from '../../types';
import styles from './WatchlistButton.module.scss';

interface WatchlistButtonProps {
  movie: Movie;
  style?: 'big' | 'small';
  user: User;
}

const Button: FC<WatchlistButtonProps> = ({ movie, style = 'small', user }) => {
  const movieId = parseInt(movie._id.toString(), 10);
  const baseVariables = { movieId: movieId, userId: user.id };

  const [isInWatchlist, setIsInWatchlist] = useState<boolean>(movie.isInWatchlist ?? false);

  const [removeMovieFromWatchlist, { error: removeError }] = useMutation(REMOVE_MOVIE_FROM_WATCHLIST, {
    variables: baseVariables,
    update(cache) {
      cache.modify({
        fields: {
          getMovieById(existingMovie) {
            if (existingMovie._id === movieId) {
              return { ...existingMovie, isInWatchlist: false }; // Update the cache to reflect the change in the UI
            }
          },
        },
      });
    },
  });

  const [addMovieToWatchlist, { error: addError }] = useMutation(ADD_MOVIE_TO_WATCHLIST, {
    variables: baseVariables,
    update(cache) {
      cache.modify({
        fields: {
          getMovieById(existingMovie) {
            if (existingMovie._id === movieId) {
              return { ...existingMovie, isInWatchlist: true }; // Update the cache to reflect the change in the UI
            }
          },
        },
      });
    },
  });

  if (removeError) {
    return <div>{removeError.message}</div>;
  }

  if (addError) {
    return <div>{addError.message}</div>;
  }

  const handleClick = async () => {
    if (isInWatchlist) {
      const response = await removeMovieFromWatchlist();
      if (response?.data?.removeMovieFromWatchlist) {
        setIsInWatchlist(false);
      }
    } else {
      const response = await addMovieToWatchlist();
      if (response?.data?.addMovieToWatchlist) {
        setIsInWatchlist(true);
      }
    }
  };

  return (
    <div className={style === 'big' ? styles.bigContainer : styles.smallContainer} data-testid='watchlist-button'>
      <ToggleButton
        value='check'
        selected={isInWatchlist}
        onChange={handleClick}
        color='primary'
        size='small'
        aria-label={isInWatchlist ? 'Remove movie from watchlist' : 'Add movie to watchlist'}
        data-testid='watchlist-toggle-button'
        title={isInWatchlist ? 'Remove movie from watchlist' : 'Add movie to watchlist'}
      >
        {isInWatchlist ? <BookmarkRemove className={styles.remove} /> : <BookmarkAdd className={styles.add} />}
        {style === 'big' && (
          <span className={styles.text}>{isInWatchlist ? 'Remove from watchlist' : 'Add to watchlist'}</span>
        )}
      </ToggleButton>
    </div>
  );
};

const WatchlistButton = memo(Button);
export default WatchlistButton;
