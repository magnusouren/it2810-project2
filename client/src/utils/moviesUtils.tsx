import { GET_MOVIES, GET_MOVIES_BY_GENRE, GET_MOVIES_BY_RATING, GET_MOVIES_BY_TITLE_AZ } from '../graphql/queries';

export const determineQueryAndVariables = (page: number, genre: string, sort: string, userId: string) => {
  const baseVariables = { page: page, userId: userId };
  const alphabeticalSort = sort === 'a-z' || sort === 'z-a';
  const ratingSort = sort === 'h-l' || sort === 'l-h';

  if (genre) {
    if (alphabeticalSort) {
      return {
        query: GET_MOVIES_BY_TITLE_AZ,
        variables: { ...baseVariables, genreId: genre, order: sort },
      };
    }
    if (ratingSort) {
      return {
        query: GET_MOVIES_BY_RATING,
        variables: { ...baseVariables, genreId: genre, order: sort },
      };
    }
    return {
      query: GET_MOVIES_BY_GENRE,
      variables: { ...baseVariables, genreId: genre },
    };
  }

  if (alphabeticalSort) {
    return {
      query: GET_MOVIES_BY_TITLE_AZ,
      variables: { ...baseVariables, order: sort },
    };
  }
  if (ratingSort) {
    return {
      query: GET_MOVIES_BY_RATING,
      variables: { ...baseVariables, order: sort },
    };
  }
  return {
    query: GET_MOVIES,
    variables: { ...baseVariables },
  };
};
