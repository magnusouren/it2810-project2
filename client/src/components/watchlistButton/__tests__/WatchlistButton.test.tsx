import { fireEvent, screen } from '@testing-library/react';

import { ADD_MOVIE_TO_WATCHLIST, REMOVE_MOVIE_FROM_WATCHLIST } from '../../../graphql/queries';
import { Movie, User } from '../../../types';
import { renderWithProviders } from '../../../utils/testUtils';
import WatchlistButton from '../WatchlistButton';

const mockUser: User = {
  id: '1234567890',
  name: 'Foo Bar',
  loginState: true,
};

const mockMovie: Movie = {
  adult: false,
  backdrop_path: '/mockBackdropURL.jpg',
  genre_ids: [
    {
      _id: 12,
      name: 'Adventure',
    },
    {
      _id: 28,
      name: 'Action',
    },
  ],
  _id: 111111,
  original_language: 'en',
  original_title: 'Foo Bar',
  overview: 'Simple mock overview',
  popularity: 1000,
  poster_path: '/mockPosterURL.jpg',
  release_date: '1970-01-01',
  title: 'Foo Bar',
  video: false,
  vote_average: 5,
  vote_count: 555,
  isInWatchlist: true,
};

describe('WatchlistButton', () => {
  it('Should match snapshot when movie is in watchlist', () => {
    const { container } = renderWithProviders(<WatchlistButton movie={mockMovie} user={mockUser} />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot when movie is not in watchlist', () => {
    const { container } = renderWithProviders(
      <WatchlistButton movie={{ ...mockMovie, isInWatchlist: false }} user={mockUser} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot when style prop is big', () => {
    const { container } = renderWithProviders(<WatchlistButton movie={mockMovie} user={mockUser} style='big' />);
    expect(container).toMatchSnapshot();
  });

  it('Should render a small button by default', () => {
    const { container } = renderWithProviders(<WatchlistButton movie={mockMovie} user={mockUser} />);
    expect(container.querySelector('.smallContainer')).toBeDefined();
  });

  it('Should call removeWatchlist mutation when button is clicked and movie is not in watchlist', async () => {
    const mocks = [
      {
        request: {
          query: REMOVE_MOVIE_FROM_WATCHLIST,
          variables: {
            movieId: 111111,
            userId: mockUser.id,
          },
        },
        result: {
          data: {
            removeMovieFromWatchlist: {
              userID: mockUser.id,
              movies: [],
            },
          },
        },
      },
    ];
    renderWithProviders(<WatchlistButton movie={mockMovie} user={mockUser} />, mocks);

    expect(screen.getByRole('button').querySelector('.remove')).toBeDefined();
    expect(screen.getByRole('button').querySelector('.add')).toBeNull();
    fireEvent.click(screen.getByRole('button'));
    expect(screen.getByRole('button').querySelector('.add')).toBeDefined();
    expect(screen.getByRole('button').querySelector('.remove')).toBeNull();
  });

  it('Should call addWatchlist mutation when button is clicked and movie is not in watchlist', async () => {
    const mocks = [
      {
        request: {
          query: ADD_MOVIE_TO_WATCHLIST,
          variables: {
            movieId: 111111,
            userId: mockUser.id,
          },
        },
        result: {
          data: {
            addMovieToWatchlist: {
              userID: mockUser.id,
              movies: [mockMovie],
            },
          },
        },
      },
    ];
    renderWithProviders(<WatchlistButton movie={{ ...mockMovie, isInWatchlist: false }} user={mockUser} />, mocks);

    expect(screen.getByRole('button').querySelector('.add')).toBeDefined();
    expect(screen.getByRole('button').querySelector('.remove')).toBeNull();
    fireEvent.click(screen.getByRole('button'));
    expect(screen.getByRole('button').querySelector('.remove')).toBeDefined();
    expect(screen.getByRole('button').querySelector('.add')).toBeNull();
  });
});
