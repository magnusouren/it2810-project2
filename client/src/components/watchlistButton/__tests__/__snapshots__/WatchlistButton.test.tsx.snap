// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WatchlistButton > Should match snapshot when movie is in watchlist 1`] = `
<div>
  <div
    class="_smallContainer_5dd437"
    data-testid="watchlist-button"
  >
    <button
      aria-label="Remove movie from watchlist"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-primary css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
      data-testid="watchlist-toggle-button"
      tabindex="0"
      title="Remove movie from watchlist"
      type="button"
      value="check"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium _remove_5dd437 css-i4bv87-MuiSvgIcon-root"
        data-testid="BookmarkRemoveIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 7h-6V5h6v2zm-2 3.9c-.32.07-.66.1-1 .1-2.76 0-5-2.24-5-5 0-1.13.37-2.16 1-3H7c-1.1 0-2 .9-2 2v16l7-3 7 3V10.9z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`WatchlistButton > Should match snapshot when movie is not in watchlist 1`] = `
<div>
  <div
    class="_smallContainer_5dd437"
    data-testid="watchlist-button"
  >
    <button
      aria-label="Add movie to watchlist"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
      data-testid="watchlist-toggle-button"
      tabindex="0"
      title="Add movie to watchlist"
      type="button"
      value="check"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium _add_5dd437 css-i4bv87-MuiSvgIcon-root"
        data-testid="BookmarkAddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 7h-2v2h-2V7h-2V5h2V3h2v2h2v2zm-2 14-7-3-7 3V5c0-1.1.9-2 2-2h7c-.63.84-1 1.87-1 3 0 2.76 2.24 5 5 5 .34 0 .68-.03 1-.1V21z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`WatchlistButton > Should match snapshot when style prop is big 1`] = `
<div>
  <div
    class="_bigContainer_5dd437"
    data-testid="watchlist-button"
  >
    <button
      aria-label="Remove movie from watchlist"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-primary css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
      data-testid="watchlist-toggle-button"
      tabindex="0"
      title="Remove movie from watchlist"
      type="button"
      value="check"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium _remove_5dd437 css-i4bv87-MuiSvgIcon-root"
        data-testid="BookmarkRemoveIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 7h-6V5h6v2zm-2 3.9c-.32.07-.66.1-1 .1-2.76 0-5-2.24-5-5 0-1.13.37-2.16 1-3H7c-1.1 0-2 .9-2 2v16l7-3 7 3V10.9z"
        />
      </svg>
      <span
        class="_text_5dd437"
      >
        Remove from watchlist
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
